src/Chapter2.hs

```diff
@@ -606,7 +630,9 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate lst = case lst of
+  (x:xs) -> x : x : duplicate xs
+  [] -> []
```

Comment:
I see that you prefer `case-of` expressions. In some cases, when the function is short, it's okay to pattern match on the top-level. Especially handy to avoid coming up with extra names for variables 🙂 
```suggestion
duplicate [] = []
duplicate (x:xs) = x : x : duplicate xs
```

Reply:
By `pattern match on the top-level` do you mean using the suggested change - I didn't fully understand how it avoids coming up with extra names for variables? Isn't the only difference going to be repeating the function name vs.  using `case varname of`?

Reply:
@saatvikshah Yes, the suggested change can be called as _pattern match on the top-level_. It avoids coming up for the variable name `lst`, becase with `case of` you need to use a variable for the list.

Generally speaking, top-level pattern matching vs case-of pattern matching is a matter of taste. I prefer to use `case-of` when the function name is long or body implementation for each case is long. However, it's not that convenient to pattern match on multiple arguments with `case-of`, so the top-level approach still can be preferred.

Anyway, use your own judgement in each case 🙂